Userland/Applications/Calendar/EventManager.cpp

```diff
@@ -97,14 +97,82 @@ ErrorOr<Vector<Event>> EventManager::deserialize_events(JsonArray const& json)
     return result;
 }
 
+Core::DateTime EventManager::format_icalendar_vevent_datetime(String const& parameter)
+{
+    auto invalid_datetime = Core::DateTime::create(0);
+    if (parameter.to_byte_string().length() < 15)
+        return invalid_datetime;
+    auto formatted_string = String::formatted("{}-{}-{}T{}:{}:{}",
+        parameter.substring_from_byte_offset(0, 4).value(),
+        parameter.substring_from_byte_offset(4, 2).value(),
+        parameter.substring_from_byte_offset(6, 2).value(),
+        parameter.substring_from_byte_offset(9, 2).value(),
+        parameter.substring_from_byte_offset(11, 2).value(),
+        parameter.substring_from_byte_offset(13, 2).value());
+    if (formatted_string.is_error())
+        return invalid_datetime;
+    auto datetime = Core::DateTime::parse(DATE_FORMAT, formatted_string.value());
+    return datetime.has_value() ? datetime.value() : invalid_datetime;
+}
+
+ErrorOr<Vector<Event>> EventManager::parse_icalendar_vevents(ByteBuffer const& content)
+{
+    Event event;
```

Comment:
Add a comment with a link to https://www.ietf.org/rfc/rfc2445.txt (…that's what we're parsing here, right?)